Bug fixes

wryonik opened the pull request:
- Add loader while account address is loading in deposits page
- Add loader and error toast in view assets page
- Fix token/NFT dropdowns in case of null list
- Update loading state in send money page

wryonik committed:
Add loader while account address is loading in deposits page

wryonik committed:
Add loader and error toast in view assets page

wryonik committed:
Fix token/NFT dropdowns in case of null list

wryonik committed:
Update loading state in send money page

Divide-By-0 committed:
make api work on render preview

Divide-By-0 committed:
remove url override since alchemy api key is set now at build time

Divide-By-0 committed:
yarn lock update

Divide-By-0 commented on app/(main)/(routes)/app/_components/Send.tsx:
```diff
@@ -18,8 +18,9 @@ const Send = () => {
-    useState<string>("TEST"); // Selected ERC20
+  const [selectedAssetString, setSelectedAssetString] = useState<
+    string | null
+  >(); // Selected ERC20
```
This fails ESlint:
```
./app/(main)/(routes)/app/_components/Send.tsx:333:27
Type error: 'selectedAssetString' is possibly 'null' or 'undefined'.
  331 |                     const assetID =
  332 |                       assetType === "ERC20"
> 333 |                         ? selectedAssetString.toString()
      |                           ^
  334 |                         : selectedNFT?.contractAddress || ""; // Using enum name instead of index
  335 |                     const recipientAddr = recipient; // Since we're sending to an email, recipient address is the email itself
  336 |                     const sendFunction =
error Command failed with exit code 1.
```

Divide-By-0 replied:
You also aren't following style convention on this line compared to the rest of the code (i.e. 3 lines instead of 1)  -- do you have some linter misconfigured or something? The error above also should have been caught by a vscode linter....

Divide-By-0 replied:
Do you have eslinter enabled? if not, can you put that in the vscode config and the precommit hook so that this doesn't happen again?

Divide-By-0 replied:
Bump @wryonik

Divide-By-0 committed:
fix selectedAssetString errors

Divide-By-0 commented:
When I click "create", it disappears the whole modal and goes to 'view assets'.

wryonik committed:
Fix .env file parsing

wryonik committed:
Fix type errors

wryonik committed:
Fix error with existing email on create new account page

This error occurred because the account code was not generated for the already existing account, but the isAccountCreated API returned true indicating the account was already created. This commit disables the flow after create account so from now on, the user have to go to login tab to login into the application instead of automatically logging in

Divide-By-0 commented on app/(main)/(routes)/app/_components/CreateAccount.tsx:
```diff
@@ -57,9 +57,9 @@ const CreateAccount: React.FC<{
-          setSelectedTab("view");
-          setSignedInState(true);
+          setStatus("Account successfully created! Please login.");
+          // setSelectedTab("send");
```
Can we set a delay then set it to switch to the login tab?

Divide-By-0 commented:
This broke state transitions in some proportion of cases in which it causes a login instead of a creation. It was reverted until this is fixed. We need an end to end test before merging any future PRs to avoid regressions.
